drivers/rtc/rtc_rv8263.c

```diff
@@ -249,6 +249,9 @@ static int rv8263c8_time_get(const struct device *dev, struct rtc_time *timeptr)
 
 	/* Return an error when the oscillator is stopped. */
 	if (regs[0] & RV8263_BM_OS) {
+		/* Clear the OS flag. */
+		regs[0] &= ~RV8263_BM_OS;
+		i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS, regs[0]);
```

Comment:
Will this not make it look like the time has been set?

Reply:
Can you please explain it further? I read all the time registers, then check if the OS bit is set, clear it and write it back and return an error from the "get" API call. The error indicates that I don´t get a valid time because the oscillator was stopped and I also clear the error condition.

Reply:
If the oscillator has stopped, that means the time (set using set_time) is no longer valid, the oscillator should be started again before set_time has been called, right? otherwise the clock is ticking, but the time is invalid :)

Reply:
If you set the time you will set the seconds and clear the OS bit if it´s set. So the time is correct. But it can happen that the OS bit is set during a get (i. e. a power cycle and reading the time).

![image](https://github.com/user-attachments/assets/0924bbe8-4e2b-41f2-8461-68d429b17490)

Reply:
The get_time API is designed to return -ENODATA if the time has not been set, which it is with the set_time API.

After a power cycle, the time is not valid, so get_time shall fail and return -ENODATA every time it is called, until time has been set using set_time:

```
int main(void)
{
        int ret;
        struct rtc_time datetime;

        /* First power on of board, RTC is not started (OS flag is set) */
        ret = rtc_get_time(rtc_dev, &datetime);
        __ASSERT(ret == -ENODATA, "time has not been set and is therefore not valid");

        /* If we try to get the time again, it should still return -ENODATA, as the time is still not valid */
        ret = rtc_get_time(rtc_dev, &datetime);
        __ASSERT(ret == -ENODATA, "time has still not been set and is therefore not valid");

        /* Now we set the time */
        ret = rtc_set_time(rtc_dev, &datetime_to_set);
        __ASSERT(ret == 0, "failed to set time");'

        /* Now, get time will work */
        ret = rtc_get_time(rtc_dev, &datetime);
        __ASSERT(ret == 0, "failed to get time");'
}
```

Reply:
Sounds good! Let me test it. Maybe we can drop this PR :)

Reply:
So it should be enough to return an error in the "get" function without clearing the OS bit. This will lead to an error during a "get" after a PoR which means you have to set the time. And this will clear the flag and the next "get" will be successful.

Reply:
That sounds exactly like what I would expect :)